pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -157,13 +158,24 @@ func newNode(name string, label map[string]string) *v1.Node {
 			Allocatable: v1.ResourceList{
 				v1.ResourcePods: resource.MustParse("100"),
 			},
+			NodeInfo: v1.NodeSystemInfo{
+				// minimum version required to use matchFields
+				KubeletVersion: "v1.11.0",
+			},
 		},
 	}
 }
 
 func addNodes(nodeStore cache.Store, startIndex, numNodes int, label map[string]string) {
```

Comment:
rather than making these a mixture of old and new nodes (which requires modifying existing tests), I'd recommend:
1. allow control of the node version being added
2. make addNodes/newNode default to adding a 1.11+ node, which should allow all existing tests to stay as-is
3. add a new test that adds some new and some old nodes, and specifically check that pods for the new nodes use MatchFields and pods for the old nodes use nodeName

Reply:
done